src/main/java/com/jcabi/http/wire/LastModifiedCachingWire.java

```diff
@@ -49,162 +41,14 @@
  * @version $Id$
  * @since 1.15
  */
-public final class LastModifiedCachingWire implements Wire {
-
-    /**
-     * Last-Modified header name.
-     */
-    public static final String LAST_MODIFIED = "Last-Modified";
-
-    /**
-     * If-Modified-Since header name.
-     */
-    private static final String IF_MODIFIED_SINCE = "If-Modified-Since";
-
-    /**
-     * Cache.
-     */
-    private final transient Map<Request, Response> cache;
-
-    /**
-     * Original wire.
-     */
-    private final transient Wire origin;
+public final class LastModifiedCachingWire
+        extends AbstractHeaderBasedCachingWire {
 
     /**
      * Public ctor.
-     * @param wire Original wire
-     */
-    public LastModifiedCachingWire(final Wire wire) {
-        this.origin = wire;
-        this.cache = new ConcurrentHashMap<>();
-    }
-
-    // @checkstyle ParameterNumber (5 lines)
-    @Override
-    public Response send(final Request req, final String home,
-        final String method,
-        final Collection<Map.Entry<String, String>> headers,
-        final InputStream content,
-        final int connect,
-        final int read) throws IOException {
-        final Response rsp;
-        if (method.equals(Request.GET)) {
-            rsp = this.consultCache(
-                req, home, method, headers, content, connect, read
-            );
-        } else {
-            rsp = this.origin.send(
-                req, home, method, headers, content, connect, read
-            );
-        }
-        return rsp;
-    }
-
-    /**
-     * Check cache and update if needed.
-     *
-     * @param req Request
-     * @param home URI to fetch
-     * @param method HTTP method
-     * @param headers Headers
-     * @param content HTTP body
-     * @param connect The connect timeout
-     * @param read The read timeout
-     * @return Response obtained
-     * @throws IOException if fails
-     * @checkstyle ParameterNumber (6 lines)
-     */
-    private Response consultCache(final Request req,
```

Comment:
@mkordas I could create new branch and PR where I could try to rename this file with git-mv